pytensor/link/pytorch/linker.py

```diff
@@ -13,7 +13,11 @@ def input_filter(self, inp: Any) -> Any:
         return pytorch_typify(inp)
 
     def output_filter(self, var: Variable, out: Any) -> Any:
-        return out.cpu()
+        import torch
+
+        if torch.is_tensor(out):
+            return out.cpu()
+        return out
```

Comment:
This whole thing is suboptimal. I think we don't need it for JAX (which returns JAX arrays/ not numpy arrays), because `np.asarray` works with it, and I guess it doesn't work for torch tensors. This should only be needed for updated shared variables where we have to convert to a common type as they could be used in multiple functions with distinct backends.

Perhaps we should expand a bit on the TorchLinker to perform the updates itself, and only force conversion when that's the case. This is already supported by `Function`.

Reply:
This is not a blocker to this PR as the problem was already here before

Reply:
Opened an issue: https://github.com/pymc-devs/pytensor/issues/1052